### PR TITLE
Topic add FLYonPIC overview picture source files

### PR DESCRIPTION
**Pull request by BrianMarre**
This PR add the source for the FLYonPIC step overview diagram and improves documentation of FLYonPIC:
- AtomicPhysicsStep_v3_1.png with svg source file
- add PlantUML source files for building blocks used in .svg source file
- updates the PlantUML documentation for better readability

### CODE REVIEW

**Comment by BrianMarre** [comment]
@psychocoderHPC as you wished

**Comment by BrianMarre** [comment]
@psychocoderHPC ready for review

File: docs/source/dev/PlantUML.rst [hunk 1]
```diff
@@ -8,22 +8,25 @@ For each UML diagram generated using PlantUML we also provide the source file ``
 
 Install PlantUML
 ----------------
-To install PlantUMl you need to
+To install PlantUMl you need to:
```
Comment (chillenzer) [inline docs/source/dev/PlantUML.rst]:
```suggestion
To install PlantUML you need to:
```
while we're at it.

File: docs/source/dev/PlantUML.rst [hunk 2]
```diff
@@ -8,22 +8,25 @@ For each UML diagram generated using PlantUML we also provide the source file ``
...
 
-- install a compatible java version
+- install a compatible ``java`` version
 - download ``plantUML.jar`` from the `PlantUML website <https://plantuml.com/download>`_ and place it in a folder of your choice.
```
Comment (chillenzer) [inline docs/source/dev/PlantUML.rst]: In general, why aren't we just linking to [this](https://plantuml.com/starting)?

**Review by chillenzer — DISMISSED** [review]
But I haven't done a diff of the two figures in detail. They look plausible from a quick glance.

File: docs/source/dev/PlantUML.rst (hunk 2 above)
Comment (chillenzer) [reply docs/source/dev/PlantUML.rst]: Even more general: [This page](https://plantuml.com/running) suggests that we could teach our docs to generate the images themselves. Might be worth an investigation.

File: docs/source/dev/PlantUML.rst (hunk 1 above)
Comment (BrianMarre) [reply docs/source/dev/PlantUML.rst]: fixed

File: docs/source/dev/PlantUML.rst (hunk 2 above)
Comment (BrianMarre) [reply docs/source/dev/PlantUML.rst]: added a link and kept the instructions as as an TLDR:

Automatically generating UML-diagrams from PlantUML source files would be nice, but I would suggest to shift this to a future PR.

**Review by chillenzer — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
failing CI tests are addressed in #5044

**Pull request merged**